tests/acceptance/test_performance_trends.py

```diff
@@ -0,0 +1,77 @@
+from sentry.utils.samples import load_data
+
+from .page_objects.base import BasePage
+
+FEATURE_NAMES = ("organizations:trends", "organizations:performance-view",)
+
+
+def make_nth_transaction(base_event, name, n, start, end):
+    event = base_event.copy()
+
+    event["transaction"] = name
+    event["event_id"] = "{:02x}".format(n).rjust(32, '0')
+    event["start_timestamp"] = iso_format(start)
+    event["timestamp"] = iso_format(end)
+
+    return event
+
+
+def make_trend(store_event, project_id, event, name, first_duration, second_duration, number_transactions=2, period_mins=60):
+    for i in range(number_transactions):
+        time_between = period_mins / number_transactions
+        minutes = period_mins - ((i + 1) * time_between) + (time_between / 2)
+        if (i < (number_transactions / 2)):
+            event_start = before_now(minutes=minutes, seconds=first_duration)
+        else:
+            event_start = before_now(minutes=minutes, seconds=second_duration)
+        event_end = before_now(minutes=minutes)
+        transaction = make_nth_transaction(event, name, i, event_start, event_end)
+        store_event(data=transaction, project_id=project_id)
+
+
+class PerformanceTrendsTest(AcceptanceTestCase, SnubaTestCase):
+    def setUp(self):
+        super(PerformanceTrendsTest, self).setUp()
+        self.org = self.create_organization(owner=self.user, name="Rowdy Tiger")
+        self.team = self.create_team(
+            organization=self.org, name="Mariachi Band", members=[self.user]
+        )
+        self.project = self.create_project(organization=self.org, teams=[self.team], name="Bengal")
+        self.group = self.create_group(project=self.project)
+        self.login_as(self.user)
+        self.path = u"/organizations/{}/performance/?{}".format(
+            self.org.slug,
+            urlencode({"view": "TRENDS", "query": "transaction.duration:>0", "statsPeriod": "1h", "project": self.project.id}),
+        )
+
+        self.page = BasePage(self.browser)
+
+    @patch("django.utils.timezone.now")
+    def test_with_data(self, mock_now):
+        mock_now.return_value = before_now().replace(tzinfo=pytz.utc)
+
+        base_event = load_data("transaction", timestamp=before_now(minutes=0))
+        make_trend(self.store_event, self.project.id, base_event, 'improvement', 2, 1)
+        make_trend(self.store_event, self.project.id, base_event, 'regression', 1, 2)
+
+        self.project.update(flags=F("flags").bitor(Project.flags.has_transactions))
+
+        with self.feature(FEATURE_NAMES):
+            self.browser.get(self.path)
+            self.page.wait_until_loaded()
+            trend_item = '[data-test-id="trends-list-item"]'
+            self.browser.wait_until(trend_item)
+            self.browser.wait_until('.echarts-for-react svg path[stroke="#4DC771"]')
+            self.browser.wait_until('.echarts-for-react svg path[stroke="#FA4747"]')
```

Comment:
This looks like it could be fragile long term, should we update our react wrappers to use echarts' render events to set a data-test-id on the chart indicating that render was completed? This could be done outside of this pull request of course.

Reply:
Yeah we could add an attr as a flag for finishing render, it's a pretty good option for this kind of thing. As for it being fragile, I know what you mean, but the reason I normally would use attrs (or classes) as flags for something like loading states is when you don't have any DOM indication that something is complete (eg. animations, hidden loading, etc) which isn't the case here, this is only fragile against someone for some reason adding another path of the same stroke colour, which I could theoretically protect against by checking against count too. Anything changing the main outcome (behaviour) of the echart for trends would cause this test to fail, which is the same fragility as any other behavioural tests, no?

That being said, I'm down for adding something to our echarts, so long as trying to determine loaded state via echarts callback is simpler than this.

Reply:
There is a [rendered](https://echarts.apache.org/en/api.html#events.rendered) event that should help.

Reply:
Does it render multiple times or just once?

Reply:
Like I was concerned about, the echarts callback just fires whenever something is rendered, which doesn't handle animation completion or progressive rendering, so could give false positives potentially. I guess as long as we don't use those then it's okay.

Reply:
We can revisit tuning the rendering detection on charts.